app/client/src/components/shared/AddSaveDataWidget.SavePanel.tsx

```diff
@@ -0,0 +1,722 @@
+
+      newSwitchesNoLayer[layer.id] = {
+        ...newSwitches[layer.id],
+        layer: null,
+      };
+    });
+
+    Object.keys(newSwitches).forEach((layerId) => {
+      if (layersOnMap.includes(layerId)) return;
+      delete newSwitches[layerId];
+    });
+
+    const oldSwitches: any = {};
+    if (saveLayersList) {
+      Object.keys(saveLayersList).forEach((key) => {
+        const value = saveLayersList[key];
+
+        oldSwitches[key] = {
+          ...value,
+          layer: null,
+        };
+      });
+    }
+
+    if (!deepEqual(newSwitchesNoLayer, oldSwitches)) {
+      setSaveLayersList(newSwitches);
+    }
+  }, [
+    layersInitialized,
+    mapLayerCount,
+    mapView,
+    saveLayersList,
+    setSaveLayersList,
+    visible,
+    widgetLayers,
+  ]);
+
+  const [publishResponse, setPublishResponse] = useState<PublishType>({
+    status: 'idle',
+  });
+
+  if (
+    !mapView ||
+    layerProps.status === 'fetching' ||
+    services.status === 'fetching'
+  ) {
+    return <LoadingSpinner />;
+  }
+
+  async function runPublish(
+    portal: __esri.Portal,
+    serviceMetaData: ServiceMetaDataType,
+    layersToPublish: LayerType[],
+  ) {
+    // check if the name is available in the user's org
+    const nameAvailableResponse = await isServiceNameAvailable(
+      portal,
+      serviceMetaData.label,
+    );
+    if (nameAvailableResponse.error) {
+      setPublishResponse({
+        status: 'failure',
+        error: {
+          message: nameAvailableResponse.error.message,
+        },
+      });
+      return;
+    }
+    if (!nameAvailableResponse.available) {
+      setPublishResponse({
+        status: 'name-not-available',
+      });
+      return;
+    }
+
+    try {
+      const response = await publish({
+        portal,
+        mapView,
+        services,
+        layers: layersToPublish,
+        layerProps,
+        serviceMetaData,
+      });
+
+      // parse the publish output
+      const layerSuccess = !response.layersResult
+        ? true
+        : response.layersResult.success;
+
+      let totalFeatures = 0;
+      let featureFailedCount = 0;
+      response.featuresResult?.forEach((l) => {
+        l.addResults?.forEach((r) => {
+          totalFeatures += 1;
+          if (!r.success) featureFailedCount += 1;
+        });
+      });
+
+      const webMapSuccess = response.webMapResult.success;
+
+      if (layerSuccess && featureFailedCount === 0 && webMapSuccess) {
+        setPublishResponse({
+          status: 'success',
+        });
+      } else {
+        let errorMessage = '';
+
+        if (!layerSuccess)
+          errorMessage +=
+            'Failed to save 1 or more layers. Check console for more information.\n';
+        if (featureFailedCount > 0)
+          errorMessage += `Partial save success. Failed to save ${featureFailedCount} of ${totalFeatures} feature(s).\n`;
+        if (webMapSuccess) errorMessage += 'Failed to save the web map.\n';
```

Comment:
It looks like this should be negated:
```suggestion
        if (!webMapSuccess) errorMessage += 'Failed to save the web map.\n';
```